Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp

```diff
@@ -84,26 +92,49 @@ struct Scene_surface_mesh_item_priv{
     idx_data_(other.d->idx_data_),
     idx_edge_data_(other.d->idx_edge_data_)
   {
-    //smesh_->collect_garbage();
     item = parent;
+    item->triangle_containers.resize(2);
+    item->edge_containers.resize(2);
+
+    item->triangle_containers[0] =
+          new Triangle_container(VI::PROGRAM_WITH_LIGHT, true);
+
+    item->triangle_containers[1] = new Triangle_container(VI::PROGRAM_WITH_LIGHT,
+                                           false);
+    item->edge_containers[0] = new Edge_container(VI::PROGRAM_WITHOUT_LIGHT,
+                                   true);
     has_feature_edges = false;
     invalidate_stats();
     vertices_displayed = true;
     edges_displayed = true;
     faces_displayed = true;
     all_primitives_displayed = false;
+    has_vcolors = false;
+    has_fcolors = false;
   }
 
   Scene_surface_mesh_item_priv(SMesh* sm, Scene_surface_mesh_item *parent):
     smesh_(sm)
   {
     item = parent;
+    item->triangle_containers.resize(2);
+    item->edge_containers.resize(2);
+    item->triangle_containers[1] = new Triangle_container(VI::PROGRAM_WITH_LIGHT,
+                                           false);
+    item->triangle_containers[0] = new Triangle_container(VI::PROGRAM_WITH_LIGHT,
+                                           true);
+    item->edge_containers[0] = new Edge_container(VI::PROGRAM_WITHOUT_LIGHT,
+                                   true);
+    item->edge_containers[1] = new Edge_container(VI::PROGRAM_NO_SELECTION,
+                                       true);
     has_feature_edges = false;
     invalidate_stats();
     vertices_displayed = true;
     edges_displayed = true;
     faces_displayed = true;
     all_primitives_displayed = false;
+    has_vcolors = false;
+    has_fcolors = false;
```

Comment:
The body of the two constructors are identical. It should be factorized into a helper method of the class.